src/spawner/spawner.py

```diff
@@ -52,37 +54,106 @@ def _make(self, data: dict) -> "SpawnerConfigurationXSchedule":
     schedule_configuration_id = ForeignKeyField(ScheduleConfiguration, null=False)
 
 
-class Spawner(Component):
-    """Abstract class for spawners. Spawners can spawn SUMO vehicles
-    based on shedules.
-    """
+class ISpawnerDisruptor(ABC):
+    """Interface for the FaultInjector to block and unblock schedules"""
 
-    @classmethod
     @abstractmethod
-    def from_json(cls, json_object: str) -> "Spawner":
-        """Constructs a spawner from a JSON object.
+    def block_schedule(self, schedule_id: str):
+        """Blocks a schedule.
 
-        :param json_object: The JSON object.
-        :type json_object: str
-        :return: A spawner.
-        :rtype: Spawner
+        :param schedule_id: The id of the schedule to block
         """
         raise NotImplementedError()
 
     @abstractmethod
-    def add_schedules_from_json(self, json_schedules: list[str]):
-        """Adds schedules to the spawner from a list of JSON objects.
+    def unblock_schedule(self, schedule_id: str):
+        """Unblocks a schedule.
 
-        :param json_schedules: The list of JSON objects.
-        :type json_schedules: list[str]
+        :param schedule_id: The id of the schedule to unblock
         """
         raise NotImplementedError()
 
-    @abstractmethod
-    def spawn_specific(self, schedule: Schedule):
-        """Spawns vehicles based on a schedule.
 
-        :param schedule: The schedule.
-        :type schedule: Schedule
+class Spawner(Component, ISpawnerDisruptor):
+    """Abstract class for spawners. Spawners can spawn SUMO vehicles
+    based on shedules.
+    """
+
+    configuration: SpawnerConfiguration
+    traci_wrapper: "ITraCiWrapper"
+    _schedules: dict[str, Schedule]
+
+    PRIOTITY: int = 0  # This will need to be set to the correct value
+
+    def next_tick(self, tick: int):
+        """Called to announce that the next tick occurred.
+
+        :param tick: The current tick.
+        :type tick: int
         """
-        raise NotImplementedError()
+        for schedule in self._schedules:
+            schedule.maybe_spawn(tick, self.traci_wrapper)
+
+    def __init__(
+        self,
+        logger: Logger,
+        configuration: SpawnerConfiguration,
+        traci_wrapper: "ITraCiWrapper",
+    ):
+        """Initializes the spawner.
+
+        :param configuration: The configuration.
+        :type configuration: SpawnerConfiguration
```

Comment:
`traci_wrapper`, `logger` is not documented :slightly_smiling_face:

Reply:
Fixed